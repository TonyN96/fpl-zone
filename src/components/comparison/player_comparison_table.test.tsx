import React from "react";
import { render, screen } from "@testing-library/react";
import { AppDataContext } from "app_content";
import { mockAppData, mockPlayers, mockPlayerStats, mockTeams } from "test/test_data";

import "@testing-library/jest-dom/extend-expect";

import { MAX_PLAYER_COUNT, PlayerComparisonTable } from ".";

describe("Player comparison table tests", () => {
  let mockSelectedPlayers = mockPlayers.slice(0, 2);

  const createComponent = (): JSX.Element => {
    return (
      <AppDataContext.Provider value={mockAppData}>
        <PlayerComparisonTable
          onAddPlayerClick={jest.fn()}
          onRemovePlayerClick={jest.fn()}
          playerStats={mockPlayerStats}
          selectedPlayers={mockSelectedPlayers}
          teams={mockTeams}
        />
      </AppDataContext.Provider>
    );
  };

  describe("Player stats", () => {
    it("Stat row rendered as expected", () => {
      render(createComponent());

      mockPlayerStats.forEach((stat) => {
        expect(screen.getByTestId(`stat-row-${stat.name}`)).toBeInTheDocument();
      });
    });

    it("Stat label cell contains label as expected", () => {
      render(createComponent());

      mockPlayerStats.forEach((stat) => {
        const labelCell = screen.getByTestId(`stat-label-cell-${stat.name}`);

        expect(labelCell).toBeInTheDocument();
        expect(labelCell).toHaveTextContent(stat.label);
      });
    });

    it("Stat value cells rendered as expected", () => {
      render(createComponent());

      mockPlayerStats.forEach((stat) => {
        mockSelectedPlayers.forEach((player) => {
          const statValueCell = screen.getByTestId(`stat-value-cell-${stat.name}-${player.id}`);

          expect(statValueCell).toBeInTheDocument();
          expect(statValueCell).toHaveTextContent(player[stat.name]);
        });
      });
    });
  });


  describe("Empty table cell", () => {
    it(`displayed when less than ${MAX_PLAYER_COUNT} players are selected`, () => {
      render(createComponent());

      mockPlayerStats.forEach((stat) => {
        expect(screen.getByTestId(`empty-table-cell=${stat.name}`)).toBeInTheDocument();
      });
    });

    it(`not displayed when ${MAX_PLAYER_COUNT} players are selected`, () => {
      mockSelectedPlayers = mockPlayers.slice(0, MAX_PLAYER_COUNT);

      render(createComponent());

      mockPlayerStats.forEach((stat) => {
        expect(screen.queryByTestId(`empty-table-cell=${stat.name}`)).toBeNull();
      });
    });
  });
});
