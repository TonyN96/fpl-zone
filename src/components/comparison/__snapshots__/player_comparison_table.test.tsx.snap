// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player comparison table tests Snapshot test 1`] = `
<DocumentFragment>
  <table
    aria-label="player comparison table"
    class="MuiTable-root css-1uzy3w7-MuiTable-root"
  >
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
        />
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium standard-table-cell css-1yhpg23-MuiTableCell-root"
        >
          <div
            class="MuiBox-root css-1bxscn6"
            data-testid="player-image-container-828"
          >
            <div
              class="MuiBox-root css-s16q1a"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-akrvbn-MuiButtonBase-root-MuiIconButton-root"
                data-testid="remove-button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium standard-table-cell css-1yhpg23-MuiTableCell-root"
        >
          <div
            class="MuiBox-root css-10izg3s"
            data-testid="player-image-container-662"
          >
            <div
              class="MuiBox-root css-s16q1a"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-akrvbn-MuiButtonBase-root-MuiIconButton-root"
                data-testid="remove-button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium standard-table-cell css-1yhpg23-MuiTableCell-root"
        >
          <div
            class="MuiBox-root css-cscpxd"
            data-testid="player-image-container-placeholder"
          >
            <div
              class="MuiBox-root css-1v6v5j6"
              data-testid="add-button"
            >
              <button
                aria-label="add player button"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AddIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Name
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="player-name-row-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Gerome Done
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="player-name-row-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Ingmar Carwithan
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Team
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="team-name-row-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Aston Villa
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="team-name-row-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Crystal Palace
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-minutes"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-minutes"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Minutes played
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-minutes-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            1451
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-minutes-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            1010
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=minutes"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-goals_scored"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-goals_scored"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Goals scored
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-goals_scored-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            2
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-goals_scored-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            19
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=goals_scored"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-assists"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-assists"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Assists
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-assists-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            7
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-assists-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            20
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=assists"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-clean_sheets"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-clean_sheets"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Clean sheets
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-clean_sheets-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            18
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-clean_sheets-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            6
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=clean_sheets"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-goals_conceded"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-goals_conceded"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Goals conceded
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-goals_conceded-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            8
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-goals_conceded-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            40
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=goals_conceded"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-own_goals"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-own_goals"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Own goals
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-own_goals-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            3
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-own_goals-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            0
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=own_goals"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-penalties_saved"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-penalties_saved"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Penalties saved
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-penalties_saved-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            0
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-penalties_saved-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            3
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=penalties_saved"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-penalties_missed"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-penalties_missed"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Penalties missed
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-penalties_missed-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            3
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-penalties_missed-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            3
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=penalties_missed"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-yellow_cards"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-yellow_cards"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Yellow cards
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-yellow_cards-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            3
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-yellow_cards-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            0
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=yellow_cards"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-red_cards"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-red_cards"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Red cards
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-red_cards-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            3
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-red_cards-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            3
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=red_cards"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-saves"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-saves"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Saves
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-saves-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            8
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-saves-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            3
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=saves"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-bonus"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-bonus"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Bonus
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-bonus-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            2
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-bonus-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            3
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=bonus"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-bps"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-bps"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Bonus Points System
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-bps-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            9
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-bps-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            38
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=bps"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-influence"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-influence"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Influence
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-influence-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            dictumst
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-influence-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            velit
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=influence"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-creativity"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-creativity"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Creativity
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-creativity-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            consectetuer
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-creativity-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            mattis
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=creativity"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-threat"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-threat"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            Threat
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-threat-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            ultrices
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-threat-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            parturient
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=threat"
        />
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        data-testid="stat-row-ict_index"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium first-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-label-cell-ict_index"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
          >
            ICT Index
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-ict_index-828"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            eu
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="stat-value-cell-ict_index-662"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            cum
          </p>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium standard-table-cell css-1ex1afd-MuiTableCell-root"
          data-testid="empty-table-cell=ict_index"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
